rmw_fastrtps_shared_cpp/include/rmw_fastrtps_shared_cpp/topic_cache.hpp

```diff
@@ -180,6 +222,7 @@ class TopicCache
         "rmw_fastrtps_shared_cpp",
         "Unable to remove topic, does not exist '%s' with type '%s'",
         topic_name.c_str(), type_name.c_str());
+      return false;
```

Comment:
@ivanpauno Added this because the current implementation seems to return `true` even if removal is not possible.